matrix function edit

osherboudara99 opened the pull request.

osherboudara99 committed:
matrix function

osherboudara99 committed:
function edit done

osherboudara99 committed:
additional function cleanup

eytan-ohana commented on Dijkstras-Algo.ipynb:
```diff
@@ -229,30 +229,31 @@
     "        print(\"-------------------------------------\")\n",
     "        for node in range(len(A)): \n",
-    "            print (node, \"\\t\\t\", distance[node]) \n",
+    "            print (node, \"\\t\\t\", nodes[node]) \n",
```
can you remove the space between print and (

eytan-ohana commented:
@osherboudara99 it looks good :+1:  
* can you use the function i made to display a path between your sources and some other node in the graph.
* for directed paths or nodes where there is no path can we add some sort of way to indicate this
      - maybe we need to change the shortest path function or even the output of dijkstra matrix/list

osherboudara99 commented:
*initializing prev in node dict to None and source vertex prev initialized to source vertex
* use shortest path function after each graph
* create function for path of source to specified node
* create function to print node dictionary in table

osherboudara99 committed:
prev initalized to none

osherboudara99 committed:
printing path function created

osherboudara99 committed:
dict table function created

eytanohana committed:
major refactors

differentiate between source and unreachable vertices\nmodifed shortest path to return src ,target , path \nmodified print path to say when there is no path

eytan-ohana commented:
Closing since there was an error with git fixed and merged locally.
